apiserver/facades/agent/upgrader/upgrader_test.go

```diff
@@ -34,17 +36,22 @@ type upgraderSuite struct {
 
 	// These are raw State objects. Use them for setup and assertions, but
 	// should never be touched by the API calls themselves
-	rawMachine *state.Machine
-	apiMachine *state.Machine
-	upgrader   *upgrader.UpgraderAPI
-	resources  *common.Resources
-	authorizer apiservertesting.FakeAuthorizer
-	hosted     *state.State
+	rawMachine             *state.Machine
+	apiMachine             *state.Machine
+	upgrader               *upgrader.UpgraderAPI
+	resources              *common.Resources
+	authorizer             apiservertesting.FakeAuthorizer
+	hosted                 *state.State
+	controllerConfigGetter *mocks.MockControllerConfigGetter
 }
 
 var _ = gc.Suite(&upgraderSuite{})
 
 func (s *upgraderSuite) SetUpTest(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	s.controllerConfigGetter = mocks.NewMockControllerConfigGetter(ctrl)
```

Comment:
We can't officially do this here as the `gc.C` isn't the same one we pass into the test. This causes much-maligned pain, but the fact is that the error from the test will be incorrect. The gomock controller will be finished at the end of the setup test, not at the end of the test. We need to move this down per test case.